base_mixin_restrict_field_access/models/restrict_field_access_mixin.py

```diff
@@ -72,6 +72,61 @@ def read(self, fields=None, load='_classic_read'):
                         self._fields[field].null(self.env))
         return result
 
+    def read_group(self, cr, uid, domain, fields, groupby, offset=0, limit=None, context=None, orderby=False, lazy=True):
+        """
+        Remove inaccessible fields from 'fields', 'groupby' and 'orderby'.
+
+        If this removes all 'fields', return no records.
+        If this removes all 'groupby', group by first remaining field.
+        If this removes 'orderby', don't specify order.
+        """
+        sanitised_fields = [f for f in fields if self._restrict_field_access_is_field_accessible(cr, uid, [], f)]
+        if 'restrict_field_access' in sanitised_fields:
+            sanitised_fields.remove('restrict_field_access')
+        if not sanitised_fields:
+            return []
+
+        sanitised_groupby = []
+        groupby = [groupby] if isinstance(groupby, basestring) else groupby
+        for groupby_part in groupby:
+            groupby_field = groupby_part.split(':')[0]
+            if self._restrict_field_access_is_field_accessible(cr, uid, [], groupby_field):
+                sanitised_groupby.append(groupby_part)
+        if not sanitised_groupby:
```

Comment:
changing the groupbys of course changes the shape of the result set, which might also cause trouble in calling code. If you go down the horribly slow route described above, you'd be better off ditching rows where a record with the groupby field inaccessible is involved